### PR TITLE
Use File instead of IO for file operations

### PR DESCRIPTION
**Pull request by Rylan12**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

See https://github.com/Homebrew/brew/pull/11140

Replace `IO.write` and `IO.read` with `File.write` and `File.read`, respectively, in `j`, `minecraft-server`, and `openzfs`.

Note that I was unable to test this for `openzfs` because it depends on certain macOS versions that I don't have access to (i.e. doesn't work on Big Sur).

### CODE REVIEW

**Comment by Rylan12** [comment]
CC @samford because I'm not sure about some of these `livecheck` blocks (especially `j`)

**Review by vitorgalvao — DISMISSED** [review]

File: Casks/minecraft-server.rb [hunk 1]
```diff
@@ -4,10 +4,18 @@
...
-  appcast "https://minecraft.net/en-us/download/server/"
   name "Minecraft Server"
+  desc "Run a Minecraft multiplayer server"
   homepage "https://minecraft.net/"
 
+  livecheck do
+    url "https://minecraft.net/en-us/download/server/"
+    strategy :page_match do |page|
+      match = page.match(%r{href=.*?/objects/(\h+)/server\.jar[^>]*>minecraft[_-]server[._-]v?(\d+(?:\.\d+)*).jar}i)
+      "#{match[2]},#{match[1]}"
```
Comment (samford) [inline Casks/minecraft-server.rb:15]:
```suggestion
    url "https://minecraft.net/en-us/download/server/"
    strategy :page_match do |page|
      page.scan(%r{href=.*?/objects/(\h+)/server\.jar[^>]*>minecraft[_-]server[._-]v?(\d+(?:\.\d+)*)\.jar}i)
          .map do |match|
        "#{match[1]},#{match[0]}"
```

The `minecraft.net` URL redirects to `www.minecraft.net`, so we can update the URL to avoid the redirection.

In general, I would say it's better to always use `page.scan(regex).map do |match|` instead of `match = page.match(regex)`, as `#scan` will match all occurrences of a regex but `#match` only matches the first occurrence. `PageMatch` uses `#scan` internally and we simply leave it up to livecheck to go through the matched versions and determine which is newest (using `Version` comparison).

Basically, `#scan` will continue to work properly if the page is updated to list more than one version whereas `#match` could potentially return an incorrect result. It's a small change but it can help reduce the maintenance burden over time.

File: Casks/openzfs.rb [hunk 2]
```diff
@@ -4,8 +4,17 @@
...
   url "https://openzfsonosx.org/w/images/#{version.after_comma[0]}/#{version.after_comma}/OpenZFS_on_OS_X_#{version.before_comma}.dmg"
   name "OpenZFS on OS X"
+  desc "Open source port of OpenZFS"
   homepage "https://openzfsonosx.org/"
 
+  livecheck do
+    url "https://openzfsonosx.org/wiki/Downloads"
+    strategy :page_match do |page|
+      match = page.match(%r{href=['"]?/w/images/\d/(\h+)/openzfs[_-]on[_-]os[_-]x[._-](\d+(?:\.\d+)*)\.dmg}i)
+      "#{match[2]},#{match[1]}"
```
Comment (samford) [inline Casks/openzfs.rb:14]:
```suggestion
      page.scan(%r{href=['"]?/w/images/\d/(\h+)/openzfs[_-]on[_-]os[_-]x[._-](\d+(?:\.\d+)*)\.dmg}i).map do |match|
        "#{match[1]},#{match[0]}"
      end
```

As above, this suggestion uses `#scan` instead of `#match`.

**Review by samford — CHANGES_REQUESTED** [review]

File: Casks/j.rb [hunk 3]
```diff
@@ -4,13 +4,19 @@
...
   homepage "https://www.jsoftware.com/"
 
   apps = %w[jbrk jcon jhs jqt]
   apps.each do |a|
     app "j#{version}/#{a}.app"
   end
 
+  livecheck do
+    url "https://github.com/jsoftware/jsource.git"
+    regex(/^jv?(\d+(?:\.\d+)*)-release$/i)
```
[the remark this answers is not in the document]
Comment (Rylan12) [reply Casks/j.rb:17]: Yeah, this is much better. I missed the fact that the `ReleaseNotes` pages were different from the `Installation` pages so I figured that wouldn't work. Glad it does though, this is much better 👍

File: Casks/minecraft-server.rb (hunk 1 above)
Comment (samford) [reply Casks/minecraft-server.rb]: I usually check the `homepage`, `stable`/`url`, etc. when I find a redirection somewhere in a formula/cask and update any URLs that are redirecting. In this case, the `homepage` also redirects to `https://www.minecraft.net/`, so we may want to update that while we're at it.

Comment (Rylan12) [reply Casks/minecraft-server.rb]: I'm happy to do whatever you prefer but the reason I split it up was because `brew style` complained that the line was too long.

**Review by Bo98 — DISMISSED** [review]

**Comment by Bo98** [comment]
Thanks!

File: Casks/minecraft-server.rb (hunk 1 above)
Comment (samford) [reply Casks/minecraft-server.rb]: It may be easy to overlook but I addressed the line length issue in the suggestion above by having `.map` on a separate line (indented to align with `.scan`). This is what the current `PageMatch` `strategy` block example in the `brew livecheck` documentation (following the homebrew/cask style) does to work around longer regexes that go past the line length limit.

Comment (Rylan12) [reply Casks/minecraft-server.rb]: Ah, gotcha 👍

**Review by SMillerDev — APPROVED** [review]
